Setting the expiration date for ibm_sm_private_certificate

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform IBM Provider Version

Terraform v1.5.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/local v2.2.3
+ provider registry.terraform.io/ibm-cloud/ibm v1.56.0

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_sm_private_certificate

### Terraform Configuration Files

```
resource "ibm_resource_instance" "secrets_manager" {
  name              = "${var.prefix}-vault"
  service           = "secrets-manager"
  plan              = "standard"
  location          = var.region
  resource_group_id = data.ibm_resource_group.resource_group.id
}

resource "ibm_sm_private_certificate_configuration_root_ca" "root_ca" {
  instance_id                       = ibm_resource_instance.secrets_manager.guid
  name                              = "${var.prefix}-root-ca"
  common_name                       = "${var.prefix}-root-ca"
  country                           = ["DE"]
  crl_disable                       = false
  crl_distribution_points_encoded   = true
  issuing_certificates_urls_encoded = true
  key_type                          = "rsa"
  key_bits                          = 4096
  max_path_length                   = 1
  max_ttl                           = "8760h"
  organization                      = ["Cool Beans GmbH"]
}

resource "ibm_sm_private_certificate_configuration_intermediate_ca" "intermidiate_ca" {
  instance_id                       = ibm_resource_instance.secrets_manager.guid
  issuer                            = ibm_sm_private_certificate_configuration_root_ca.root_ca.name
  name                              = "${var.prefix}-intermediate-ca"
  common_name                       = "${var.prefix}-intermediate-ca"
  country                           = ["DE"]
  crl_disable                       = false
  crl_distribution_points_encoded   = true
  issuing_certificates_urls_encoded = true
  key_type                          = "rsa"
  key_bits                          = 2048
  max_path_length                   = 0
  max_ttl                           = "8760h"
  organization                      = ["Cool Beans GmbH"]
  signing_method                    = "internal"
}

resource "ibm_sm_private_certificate_configuration_template" "certificate_template" {
  instance_id           = ibm_resource_instance.secrets_manager.guid
  certificate_authority = ibm_sm_private_certificate_configuration_intermediate_ca.intermidiate_ca.name
  name                  = "${var.prefix}-certificate-template"
  country               = ["DE"]
  key_type              = "rsa"
  key_bits              = 2048
  max_ttl               = "8760h"
  organization          = ["Cool Beans GmbH"]
  allow_any_name        = true
}

resource "ibm_sm_private_certificate" "vpn_server" {
  instance_id          = ibm_resource_instance.secrets_manager.guid
  name                 = "${var.prefix}-vpn-server"
  certificate_template = ibm_sm_private_certificate_configuration_template.certificate_template.name
  common_name          = "${var.prefix}-vpn-server"
  expiration_date      = "2023-09-16T12:10:20Z"
}
```

### Debug Output

```text
➜  k8s-playground git:(master) ✗ export TF_LOG=ERROR
➜  k8s-playground git:(master) ✗ terraform validate 
2023-08-15T14:46:34.751+0200 [ERROR] provider.terraform-provider-ibm_v1.56.0: Response contains error diagnostic: @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_summary="Value for unconfigurable attribute" tf_proto_version=5.3 tf_provider_addr=provider tf_resource_type=ibm_sm_private_certificate diagnostic_attribute=AttributeName("expiration_date") tf_rpc=ValidateResourceTypeConfig @module=sdk.proto diagnostic_detail="Can't configure a value for "expiration_date": its value will be decided automatically based on the result of applying this configuration." diagnostic_severity=ERROR tf_req_id=5353eda7-c148-77c2-6e8e-4ef3d2205a5d timestamp=2023-08-15T14:46:34.751+0200
2023-08-15T14:46:34.751+0200 [ERROR] vertex "ibm_sm_private_certificate.vpn_server" error: Value for unconfigurable attribute
╷
│ Error: Value for unconfigurable attribute
│ 
│   with ibm_sm_private_certificate.vpn_server,
│   on secrets-manager.tf line 58, in resource "ibm_sm_private_certificate" "vpn_server":
│   58:   expiration_date      = "2023-09-16T12:10:20Z"
│ 
│ Can't configure a value for "expiration_date": its value will be decided automatically based on the result of applying this configuration.
```

### Expected Behavior

I would expect that the expiration date to be set. The documentation says that you should be able to do this.

### Actual Behavior

The configuration does not validate. If I leave out the `expiration_date` then it is set for 31 days. I would like it to be 90 days.

### Steps to Reproduce

1. `terraform validate`

### References

https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/sm_private_certificate#expiration_date

--- a/website/docs/r/sm_private_certificate.html.markdown
+++ b/website/docs/r/sm_private_certificate.html.markdown
@@ -28,6 +28,7 @@ resource "ibm_sm_private_certificate" "sm_private_certificate"{
 		unit = "day"
   }
   secret_group_id = ibm_sm_secret_group.sm_secret_group.secret_group_id
+  ttl = "48h"
 }
 ```
 
@@ -46,21 +47,21 @@ Review the argument reference that you can specify for your resource.
 * `custom_metadata` - (Optional, Map) The secret metadata that a user can customize.
 * `description` - (Optional, String) An extended description of your secret.To protect your privacy, do not use personal data, such as your name or location, as a description for your secret group.
   * Constraints: The maximum length is `1024` characters. The minimum length is `0` characters. The value must match regular expression `/(.*?)/`.
-* `expiration_date` - (Optional, Forces new resource, String) The date a secret is expired. The date format follows RFC 3339.
 * `labels` - (Optional, List) Labels that you can use to search for secrets in your instance.Up to 30 labels can be created.
   * Constraints: The list items must match regular expression `/(.*?)/`. The maximum length is `30` items. The minimum length is `0` items.
 * `name` - (Required, String) The human-readable name of your secret.
     * Constraints: The maximum length is `256` characters. The minimum length is `2` characters. The value must match regular expression `^[A-Za-z0-9][A-Za-z0-9]*(?:_*-*\\.*[A-Za-z0-9]+)*$`.
 * `rotation` - (Optional, List) Determines whether Secrets Manager rotates your secrets automatically.
 Nested scheme for **rotation**:
-	* `auto_rotate` - (Optional, Boolean) Determines whether Secrets Manager rotates your secret automatically.Default is `false`. If `auto_rotate` is set to `true` the service rotates your secret based on the defined interval.
-	* `interval` - (Optional, Integer) The length of the secret rotation time interval.
-	  * Constraints: The minimum value is `1`.
-	* `rotate_keys` - (Optional, Boolean) Determines whether Secrets Manager rotates the private key for your public certificate automatically.Default is `false`. If it is set to `true`, the service generates and stores a new private key for your rotated certificate.
-	* `unit` - (Optional, String) The units for the secret rotation time interval.
-	  * Constraints: Allowable values are: `day`, `month`.
+    * `auto_rotate` - (Optional, Boolean) Determines whether Secrets Manager rotates your secret automatically.Default is `false`. If `auto_rotate` is set to `true` the service rotates your secret based on the defined interval.
+    * `interval` - (Optional, Integer) The length of the secret rotation time interval.
+      * Constraints: The minimum value is `1`.
+    * `rotate_keys` - (Optional, Boolean) Determines whether Secrets Manager rotates the private key for your public certificate automatically.Default is `false`. If it is set to `true`, the service generates and stores a new private key for your rotated certificate.
+    * `unit` - (Optional, String) The units for the secret rotation time interval.
+      * Constraints: Allowable values are: `day`, `month`.
 * `secret_group_id` - (Optional, Forces new resource, String) A v4 UUID identifier, or `default` secret group.
   * Constraints: The maximum length is `36` characters. The minimum length is `7` characters. The value must match regular expression `/^([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}|default)$/`.
+* `ttl` - (Optional, Forces new resource, String) The time-to-live (TTL) to assign to the private certificate. The value can be supplied as a string duration with time unit suffix - `d` for days, `h` for hours, `m` for minutes, or `s` for seconds. For example, `2d` or `48h` or `172800s`.
 
 ## Attribute Reference
 
@@ -80,6 +81,7 @@ In addition to all argument references listed, you can access the following attr
 * `crn` - (String) A CRN that uniquely identifies an IBM Cloud resource.
   * Constraints: The maximum length is `512` characters. The minimum length is `9` characters. The value must match regular expression `/^crn:v[0-9](:([A-Za-z0-9-._~!$&'()*+,;=@\/]|%[0-9A-Z]{2})*){8}$/`.
 * `downloaded` - (Boolean) Indicates whether the secret data that is associated with a secret version was retrieved in a call to the service API.
+* `expiration_date` - (String) The date the certificate is expired. The date format follows RFC 3339.
 * `issuer` - (Forces new resource, String) The distinguished name that identifies the entity that signed and issued the certificate.
   * Constraints: The maximum length is `128` characters. The minimum length is `2` characters. The value must match regular expression `/(.*?)/`.
 * `issuing_ca` - (String) The PEM-encoded certificate of the certificate authority that signed and issued this certificate.

